controlplane/eks/api/v1alpha4/awsmanagedcontrolplane_types.go

```diff
@@ -173,6 +173,12 @@ type EndpointAccess struct {
 	// Private points VPC-internal control plane access to the private endpoint
 	// +optional
 	Private *bool `json:"private,omitempty"`
+	// BastionAccess controls whether API access is available from the bastion host.
+	// +optional
+	BastionAccess *bool `json:"bastionAccess,omitempty"`
+	// AdditionalSecurityGroups specifies additional securirty groups that should be granted API access.
+	// +optional
+	AdditionalSecurityGroups []*string `json:"additionalSecurityGroups,omitempty"`
```

Comment:
We are moving away from a slice of pointers to structs due to an issue with conversion-gen (see #2495). Could you change to `[]string` and then check for the length of this elsewhere.